lib/puppet/provider/package/pip.rb

```diff
@@ -226,6 +277,28 @@ def install_options
     join_options(@resource[:install_options])
   end
 
+  def insync?(is)
+    return false unless is && is != :absent
+
+    should = @resource[:ensure]
+
+    begin
```

Comment:
Similar to my comment above, I would preemptively check if `is` and `should` match `/<>=/` before doing version range stuff.